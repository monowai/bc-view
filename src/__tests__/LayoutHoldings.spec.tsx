import React from "react";
import { cleanup, render, screen, waitForElementToBeRemoved } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import ViewHoldings from "../holdings";
import nock from "nock";
import { MemoryRouter } from "react-router";

afterEach(cleanup);
afterAll(nock.cleanAll);

jest.mock("@react-keycloak/ssr", () => ({
  useKeycloak: () => ({
    initialized: true,
    keycloak: { token: "abc" },
  }),
}));

nock("http://localhost", {
  reqheaders: {
    Authorization: "Bearer abc",
  },
})
  .persist(true)
  .get("/bff/test/today")
  .replyWithFile(200, __dirname + "/__contracts__/test-holdings.json", {
    "Access-Control-Allow-Origin": "*",
    "Content-type": "application/json",
  })
  .get("/bff/zero/today")
  .replyWithFile(200, __dirname + "/__contracts__/zero-holdings.json", {
    "Access-Control-Allow-Origin": "*",
    "Content-type": "application/json",
  });

describe("<ViewHoldings />", () => {
  it("matches snapshot when getData present", async () => {
    const TestHoldings = (): JSX.Element => {
      return ViewHoldings("test");
    };
    const { container } = render(
      <MemoryRouter initialEntries={["/portfolios"]} keyLength={0}>
        <TestHoldings />
      </MemoryRouter>
    );
    await screen.findByTestId("loading");
    await waitForElementToBeRemoved(() => screen.getByTestId("loading"));
    expect(nock.isDone());
    await screen.findByText("USD");
    expect(container).toMatchSnapshot();
  });

  it("matches snapshot for zero getData", async () => {
    const ZeroHoldings = (): JSX.Element => {
      return ViewHoldings("zero");
    };
    const { container } = render(<ZeroHoldings />);
    expect(nock.isDone());
    await screen.findByTestId("dropzone");
    expect(container).toMatchSnapshot();
  });
});
