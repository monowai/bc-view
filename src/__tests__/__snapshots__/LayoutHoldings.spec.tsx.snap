// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewHoldings /> matches snapshot for zero getData 1`] = `
<div>
  <div
    data-testid="dropzone"
  >
    <label>
      This portfolio has no transactions. Please drop your CSV file to upload
    </label>
    <div
      tabindex="0"
    >
      <input
        autocomplete="off"
        multiple=""
        style="display: none;"
        tabindex="-1"
        type="file"
      />
      <span>
        <i
          class="far fa-arrow-alt-circle-up fa-3x"
        />
      </span>
    </div>
  </div>
</div>
`;

exports[`<ViewHoldings /> matches snapshot when getData present 1`] = `
<div>
  <div
    class="page-box"
  >
    <div
      class="filter-columns"
    >
      <div
        class="filter-label"
      >
        Value In
      </div>
      <div
        class="filter-column"
      >
        <div
          class=" css-2b097c-container"
        >
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class=" css-yk16xz-control"
          >
            <div
              class=" css-g1d714-ValueContainer"
            >
              <div
                class=" css-1uccc91-singleValue"
              >
                Portfolio
              </div>
              <input
                aria-autocomplete="list"
                class="css-wmatm6-dummyInput-DummyInput"
                id="react-select-2-input"
                readonly=""
                tabindex="0"
                value=""
              />
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="filter-label"
      >
        Group By
      </div>
      <div
        class="filter-column"
      >
        <div
          class=" css-2b097c-container"
        >
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class=" css-yk16xz-control"
          >
            <div
              class=" css-g1d714-ValueContainer"
            >
              <div
                class=" css-1uccc91-singleValue"
              >
                Currency
              </div>
              <input
                aria-autocomplete="list"
                class="css-wmatm6-dummyInput-DummyInput"
                id="react-select-3-input"
                readonly=""
                tabindex="0"
                value=""
              />
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="filter-label"
      >
        Open Only
      </div>
      <div
        class="filter-column"
      >
        <div
          class="react-switch"
          style="position: relative; display: inline-block; text-align: left; opacity: 1; direction: ltr; border-radius: 14px; transition: opacity 0.25s; user-select: none;"
        >
          <div
            class="react-switch-bg"
            style="height: 28px; width: 56px; margin: 0px; position: relative; background: rgb(0, 136, 0); border-radius: 14px; cursor: pointer; transition: background 0.25s;"
          >
            <div
              style="height: 28px; width: 30px; position: relative; opacity: 1; pointer-events: none; transition: opacity 0.25s;"
            >
              <svg
                height="100%"
                style="position: absolute; top: 0px;"
                viewBox="-2 -5 17 21"
                width="100%"
              >
                <path
                  d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
                  fill="#fff"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <div
              style="height: 28px; width: 30px; position: absolute; opacity: 0; right: 0px; top: 0px; pointer-events: none; transition: opacity 0.25s;"
            >
              <svg
                height="100%"
                style="position: absolute; top: 0px;"
                viewBox="-2 -5 14 20"
                width="100%"
              >
                <path
                  d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
                  fill="#fff"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </div>
          <div
            class="react-switch-handle"
            style="height: 26px; width: 26px; background: rgb(255, 255, 255); display: inline-block; cursor: pointer; border-radius: 50%; position: absolute; transform: translateX(29px); top: 1px; outline: 0; border: 0px; transition: background-color 0.25s, transform 0.25s, box-shadow 0.15s;"
          />
          <input
            checked=""
            required=""
            role="switch"
            style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <div
      class="stats-container"
    >
      <table>
        <tbody>
          <tr
            class="stats-header"
          >
            <th
              align="left"
            >
              Summary
            </th>
            <th
              align="right"
            />
            <th
              align="right"
            />
            <th
              align="right"
            />
            <th
              align="right"
            />
            <th
              align="right"
            />
          </tr>
        </tbody>
        <tbody>
          <tr
            class="stats-row"
          >
            <td>
              <div
                class="left-cell"
              >
                <a
                  href="/portfolios/test"
                >
                  <span
                    class="has-tooltip-right"
                    data-tooltip="Test Portfolio"
                  >
                    TEST
                     
                    : 
                  </span>
                </a>
                NZD
              </div>
            </td>
            <td
              align="right"
            >
              <span>
                -
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                8,478.00
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                2,729.71
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                2,961.31
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                -2,786.98
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="all-getData"
    >
      <table
        class="table is-striped is-hoverable"
      >
        <tbody
          class="table-header"
        >
          <tr>
            <th>
              AUD
            </th>
            <th
              align="right"
            />
            <th
              align="right"
            />
            <th
              align="right"
            />
            <th
              align="right"
            />
            <th
              align="right"
            />
            <th
              align="right"
            />
            <th
              align="right"
            />
            <th
              align="right"
            />
            <th
              align="right"
            />
            <th
              align="right"
            />
            <th
              align="right"
            />
          </tr>
        </tbody>
        <tbody>
          <tr
            class="holding-row"
          >
            <td
              class="asset"
            >
              AMP: Amp Limited
            </td>
            <td
              align="right"
              class="price"
            >
              <span
                data-tooltip="2020-05-01"
              >
                $
                <span>
                  132.03
                </span>
              </span>
            </td>
            <td
              align="right"
            >
              <span
                class="negative-gain"
                data-tooltip="Previous $ 135.35"
              >
                -2.51
                %
              </span>
            </td>
            <td
              align="right"
            >
              <a
                href="/trns/test/asset/0OKznUxcT9WlNWIAp2dn2Q/trades"
              >
                <span>
                  637
                </span>
              </a>
            </td>
            <td
              align="right"
            >
              <span>
                -
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                -2,436.73
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                -
              </span>
              %
            </td>
            <td
              align="right"
            >
              <span>
                2,436.73
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                3.83
              </span>
            </td>
            <td
              align="right"
            >
              <span
                data-tooltip="Last Event: undefined"
              >
                <a
                  href="/trns/test/asset/0OKznUxcT9WlNWIAp2dn2Q/events"
                >
                  <span>
                    2,598.62
                  </span>
                </a>
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                -
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                161.89
              </span>
            </td>
          </tr>
        </tbody>
        <tbody
          class="holding-totals-row"
        >
          <tr
            class="holding-footer"
          >
            <td
              align="right"
              colspan="4"
            >
              Sub-Total - 
              NZD
            </td>
            <td
              align="right"
            >
              <span>
                -
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                -2,436.73
              </span>
            </td>
            <td />
            <td
              align="right"
            >
              <span>
                2,436.73
              </span>
            </td>
            <td />
            <td
              align="right"
            >
              <span>
                2,598.62
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                -
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                161.89
              </span>
            </td>
          </tr>
        </tbody>
        <tbody
          class="table-header"
        >
          <tr>
            <th>
              USD
            </th>
            <th
              align="right"
            />
            <th
              align="right"
            />
            <th
              align="right"
            />
            <th
              align="right"
            />
            <th
              align="right"
            />
            <th
              align="right"
            />
            <th
              align="right"
            />
            <th
              align="right"
            />
            <th
              align="right"
            />
            <th
              align="right"
            />
            <th
              align="right"
            />
          </tr>
        </tbody>
        <tbody>
          <tr
            class="holding-row"
          >
            <td
              class="asset"
            >
              AAPL: Apple Inc
            </td>
            <td
              align="right"
              class="price"
            >
              <span
                data-tooltip="2020-05-01"
              >
                $
                <span>
                  132.03
                </span>
              </span>
            </td>
            <td
              align="right"
            >
              <span
                class="negative-gain"
                data-tooltip="Previous $ 135.35"
              >
                -2.51
                %
              </span>
            </td>
            <td
              align="right"
            >
              <a
                href="/trns/test/asset/pw1dIdlhSx2x8tnlA2qNhw/trades"
              >
                <span>
                  12
                </span>
              </a>
            </td>
            <td
              align="right"
            >
              <span>
                -
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                -1,436.44
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                -
              </span>
              %
            </td>
            <td
              align="right"
            >
              <span>
                1,436.44
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                119.70
              </span>
            </td>
            <td
              align="right"
            >
              <span
                data-tooltip="Last Event: undefined"
              >
                <a
                  href="/trns/test/asset/pw1dIdlhSx2x8tnlA2qNhw/events"
                >
                  <span>
                    126.33
                  </span>
                </a>
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                132.11
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                -1,178.00
              </span>
            </td>
          </tr>
          <tr
            class="holding-row"
          >
            <td
              class="asset"
            >
              ABC: Amerisource Bergen Corp.
            </td>
            <td
              align="right"
              class="price"
            >
              <span
                data-tooltip="2020-05-01"
              >
                $
                <span>
                  132.03
                </span>
              </span>
            </td>
            <td
              align="right"
            >
              <span
                class="negative-gain"
                data-tooltip="Previous $ 135.35"
              >
                -2.51
                %
              </span>
            </td>
            <td
              align="right"
            >
              <a
                href="/trns/test/asset/QyZk-QioSRqyq1alwSf5Tw/trades"
              >
                <span>
                  25
                </span>
              </a>
            </td>
            <td
              align="right"
            >
              <span>
                -
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                -1,263.00
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                -
              </span>
              %
            </td>
            <td
              align="right"
            >
              <span>
                1,263.00
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                50.52
              </span>
            </td>
            <td
              align="right"
            >
              <span
                data-tooltip="Last Event: undefined"
              >
                <a
                  href="/trns/test/asset/QyZk-QioSRqyq1alwSf5Tw/events"
                >
                  <span>
                    6.50
                  </span>
                </a>
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                -
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                -1,256.50
              </span>
            </td>
          </tr>
          <tr
            class="holding-row"
          >
            <td
              class="asset"
            >
              CSCO: Cisco Systems, Inc.
            </td>
            <td
              align="right"
              class="price"
            >
              <span
                data-tooltip="2020-05-01"
              >
                $
                <span>
                  132.03
                </span>
              </span>
            </td>
            <td
              align="right"
            >
              <span
                class="negative-gain"
                data-tooltip="Previous $ 135.35"
              >
                -2.51
                %
              </span>
            </td>
            <td
              align="right"
            >
              <a
                href="/trns/test/asset/HQuFW0B9RUW9bML5r2x0pA/trades"
              >
                <span>
                  55
                </span>
              </a>
            </td>
            <td
              align="right"
            >
              <span>
                -
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                -971.52
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                -
              </span>
              %
            </td>
            <td
              align="right"
            >
              <span>
                971.52
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                17.66
              </span>
            </td>
            <td
              align="right"
            >
              <span
                data-tooltip="Last Event: undefined"
              >
                <a
                  href="/trns/test/asset/HQuFW0B9RUW9bML5r2x0pA/events"
                >
                  <span>
                    229.86
                  </span>
                </a>
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                227.29
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                -514.37
              </span>
            </td>
          </tr>
        </tbody>
        <tbody
          class="holding-totals-row"
        >
          <tr
            class="holding-footer"
          >
            <td
              align="right"
              colspan="4"
            >
              Sub-Total - 
              NZD
            </td>
            <td
              align="right"
            >
              <span>
                -
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                -3,670.96
              </span>
            </td>
            <td />
            <td
              align="right"
            >
              <span>
                3,670.96
              </span>
            </td>
            <td />
            <td
              align="right"
            >
              <span>
                362.69
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                359.40
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                -2,948.87
              </span>
            </td>
          </tr>
        </tbody>
        <tbody
          class="totals-row"
        >
          <tr>
            <td
              align="right"
              colspan="4"
            >
              Totals in 
              PORTFOLIO
               currency
            </td>
            <td
              align="right"
            >
              <span>
                -
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                -6,107.69
              </span>
            </td>
            <td />
            <td
              align="right"
            >
              <span>
                6,107.69
              </span>
            </td>
            <td />
            <td
              align="right"
            >
              <span>
                2,961.31
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                359.40
              </span>
            </td>
            <td
              align="right"
            >
              <span>
                -2,786.98
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
